### ISSUE TITLE
Some scales involve the filtered nulls by `Invalid: "filter"`

### ISSUE THREAD

**Issue by yhoonkim** [vega/vega-lite#8479]
![image](https://user-images.githubusercontent.com/6269456/195919063-1672a5c3-2a8a-40d4-9700-3e252ce14b43.png)

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "description": "A simple bar chart with embedded data.",
  "data": {
    "values": [
      {"a": null, "b": 9999999},
      {"a": 10, "b": 55},
      {"a": 20, "b": 43}
    ]
  },
  "mark": {"type": "line", "invalid": "filter"},
  "encoding": {
    "x": {"field": "a", "type": "quantitative", "axis": {"labelAngle": 0}},
    "y": {"field": "b", "type": "quantitative"}
  }
}
```

` {"a": null, "b": 9999999}`  is supposed to be excluded from the visualization so that the y scale should involve the other two rows only.

**Comment by rudzimj** [comment]
I agree. It doesn't look like `"invalid": "filter"` works at all. Null and NaN should have been filtered out in this chart:

<img width="143" alt="image" src="https://user-images.githubusercontent.com/36211641/205340295-11f6bc13-896b-46f8-9379-d844d9fc2767.png">

[Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaoAMo79AAlhJZsuILQgAwopAA5BgiZwAThbYAZlYMEJhsCL4GqACMAMwAvgqm5hBWMDZ2DugAIq4eXr7+QVAhYRE+UQCcicmClta29o4Aonme3n6BwaHhkWjR1UkgZnWpDZmObkhubQWdxaW9FY4AHiA1wylpGU3oHoIOSvkdRd1lfaiyDAcbI-XpjVkgAFq+bLMnXSU95VEADPEALqJECCJAAT18RlAyB8AGs0KBMOCcHBHEwkD5XFA2D47D4AEpIZQ0EJoMR-JQ4wS4xwAYkqfycACZGa4wmxBJgaDg0JgfHolDRZNphMpHAEaFyofENrCEZoQMjUY5MHAVph2WxOdzeah+YKQDoaGR5Og6gFNUoMRB7MK0egEDRlMo6q5lGtUGIqZzaegmGCoAiQXBZDiSbIyIilerNYrJfZxXt2oUvotfuyUQ6QABHBg2bnqbmkdaywFKUPh4VRxXg6MJwRJzajbaPTMq9CycLCnSuGQ+OMqfRQUMRqNKQtuy7XQQgz2gBtN46phY-PoTrOOPMFuhqGglid0KdXA5KJArGhRE+z2VAA)

A workaround is to add a transform filter like this:
`"transform": [{"filter": {"field": "Number of customers", "valid": true}}],`

**Comment by PBI-David** [comment]
In the generated Vega spec, it seems the root cause is that the scale is being derived from source_0 instead of the filtered data_0.

**Comment by kanitw** [comment]
> I agree. It doesn't look like "invalid": "filter" works at all. Null and NaN should have been filtered out in this chart:

We do not consider NaN, Null "invalid" for discrete scales (since discrete scales can display any values).

So this is by design.

**Comment by somedaygone** [comment]
It's ok for this to be by design, but the documentation needs to match.  What you said is not what [this](https://vega.github.io/vega-lite/docs/mark.html#general) says:

> Defines how Vega-Lite should handle marks for invalid values (`null` and `NaN`).
> 
> * If set to `"filter"` (default), all data items with null values will be skipped (for line, trail, and area marks) or filtered (for other marks).
> * If `null`, all data items are included. In this case, invalid values will be interpreted as zeroes.
